server/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTests.java

```diff
@@ -345,4 +327,155 @@ protected QueryBuilder parseQuery(XContentParser parser) throws IOException {
         return query;
     }
 
+    public void testSerializationFromLowerVersion() throws IOException {
```

Comment:
That seems fragile since the serialization could change in lower version after this PR so we can diverge.
It's a good thing to test serialization in a unit test but it should be limited to the code that is implemented in this branch (without TermsQueryBuilderBeforeV8).